apps/klasses/templates/klasses/wizard/define_homework.html

```diff
@@ -43,10 +44,41 @@ <h1>Homework</h1>
                         </td>
                         <td>{{ homework_def.submissions.count }}</td>
                         <td>{{ homework_def.avg_grade }}</td>
+                        <td><a id="{{ homework_def.pk }}" class="ui red mini button delete-definition">X</a></td>
                     </tr>
                 {% endfor %}
                 </tbody>
             </table>
         </div>
     </div>
 {% endblock %}
+
+{% block extra_script %}
+    <script>
+        $(document).ready( function () {
+
+            function delete_definition(pk) {
+                CHAGRADE.api.delete_definition(pk)
+                    .done(function (data) {
+                        toastr.success("Successfully deleted homework")
```

Comment:
This `toastr` message is only visible for a fraction of a second due to the reloading of the page. Usually for situations like this, we need to either: 

1. **Dynamically update the page dynamically after deletion to eliminate our need to reload the page.**
The current form of this page doesn't lend itself to this, since it's solely rendered by the server rather than by the browser. If it were a riot tag, this would be a better option.

2. **Display a success message immediately after the reload.**
This is a better option that we can achieve by passing a query_param to our `window.location` and conditionally passing a flag in context that indicates whether we pass the query_param. Use this to conditionally trigger a `toastr` success message for homework definition deletion.

Reply:
Got an working solution.